### PR TITLE
Add backoff jitter

### LINKED ISSUE #10482: Use `backoff_jitter` when making HTTP requests
### Summary

```diff
diff --git a/mlflow/environment_variables.py b/mlflow/environment_variables.py
index 59f2ca94c..4edfe79b4 100644
--- a/mlflow/environment_variables.py
+++ b/mlflow/environment_variables.py
@@ -101,6 +101,12 @@ MLFLOW_HTTP_REQUEST_BACKOFF_FACTOR = _EnvironmentVariable(
     "MLFLOW_HTTP_REQUEST_BACKOFF_FACTOR", int, 2
 )
 
+#: Specifies the backoff jitter between MLflow HTTP request failures
+#: (default: ``1``)
+MLFLOW_HTTP_REQUEST_BACKOFF_JITTER = _EnvironmentVariable(
+    "MLFLOW_HTTP_REQUEST_BACKOFF_JITTER", float, 1.0
+)
+
 #: Specifies the timeout in seconds for MLflow HTTP requests
 #: (default: ``120``)
 MLFLOW_HTTP_REQUEST_TIMEOUT = _EnvironmentVariable("MLFLOW_HTTP_REQUEST_TIMEOUT", int, 120)
diff --git a/mlflow/utils/rest_utils.py b/mlflow/utils/rest_utils.py
index cab83d4a7..2dd1e42bc 100644
--- a/mlflow/utils/rest_utils.py
+++ b/mlflow/utils/rest_utils.py
@@ -5,6 +5,7 @@ import requests
 
 from mlflow.environment_variables import (
     MLFLOW_HTTP_REQUEST_BACKOFF_FACTOR,
+    MLFLOW_HTTP_REQUEST_BACKOFF_JITTER,
     MLFLOW_HTTP_REQUEST_MAX_RETRIES,
     MLFLOW_HTTP_REQUEST_TIMEOUT,
 )
@@ -106,6 +107,7 @@ def http_request(
             headers=headers,
             verify=host_creds.verify,
             timeout=timeout,
+            backoff_jitter=MLFLOW_HTTP_REQUEST_BACKOFF_JITTER.get(),
             **kwargs,
         )
     except requests.exceptions.Timeout as to:
```

https://aws.amazon.com/builders-library/timeouts-retries-and-backoff-with-jitter/ might be worth a read.

### Notes

- Make sure to open a PR from a **non-master** branch.
- Sign off the commit using the `-s` flag when making a commit:

  ```sh
  git commit -s -m "..."
           # ^^ make sure to use this
  ```

- Include `#{issue_number}` (e.g. `#123`) in the PR description when opening a PR.

### PR DESCRIPTION
**Pull request by ajinkyavbhandare**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/ajinkyavbhandare/mlflow/pull/10486?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10486/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10486
```

</p>
</details>

### Related Issues/PRs

Resolve #10482

### What changes are proposed in this pull request?

This PR adds a backoff jitter to HTTP requests in MLflow. This is achieved by introducing a new environment variable `MLFLOW_HTTP_REQUEST_BACKOFF_JITTER` and using this variable when making HTTP requests.

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Yes. This change introduces a backoff jitter to HTTP requests, which can help improve the efficiency and reliability of the requests.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [x] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/environment_variables.py [hunk 1]
```diff
@@ -101,6 +101,12 @@ def get(self):
     "MLFLOW_HTTP_REQUEST_BACKOFF_FACTOR", int, 2
 )
 
+#: Specifies the backoff jitter between MLflow HTTP request failures
+#: (default: ``1``)
```
Comment (harupy) [inline mlflow/environment_variables.py:105]:
```suggestion
#: (default: ``1.0``)
```

**Comment by harupy** [comment]
@ajinkyavbhandare why you closed it?

**Comment by ajinkyavbhandare** [comment]
hello @harupy ,I made some changes to the code for the and created a new pull request, #10486. Could you please review the new pull request and reassign me the task? thank you.

**Comment by harupy** [comment]
where is the new PR?

**Comment by harupy** [comment]
@ajinkyavbhandare what's the difference?

**Comment by harupy** [comment]
we can keep working on this one.

**Comment by harupy** [comment]
let me push a commit.

File: mlflow/utils/rest_utils.py [hunk 2]
```diff
@@ -60,9 +62,14 @@ def http_request(
 
     :return: requests.Response object.
     """
-    max_retries = max_retries or MLFLOW_HTTP_REQUEST_MAX_RETRIES.get()
-    backoff_factor = backoff_factor or MLFLOW_HTTP_REQUEST_BACKOFF_FACTOR.get()
-    timeout = timeout or MLFLOW_HTTP_REQUEST_TIMEOUT.get()
+    max_retries = MLFLOW_HTTP_REQUEST_MAX_RETRIES.get() if max_retries is None else max_retries
```
Comment (harupy) [inline mlflow/utils/rest_utils.py:65]: In the original code, `max_retries = 0` falls back to `MLFLOW_HTTP_REQUEST_MAX_RETRIES.get()` but it should not.

**Review by harupy — APPROVED** [review]
LGTM

**Review by smurching — APPROVED** [review]

**Comment by harupy** [comment]
Manually tested using this toy fastapi app:

```python
from fastapi import FastAPI, HTTPException, status


app = FastAPI()


counter = 0


@app.get("/test")
def test():
    global counter
    counter += 1

    if counter < 3:
        raise HTTPException(
            status_code=status.HTTP_502_BAD_GATEWAY,
            detail="Service unavailable",
        )

    else:
        # Reset counter
        counter = 0
        return {"counter": counter}


# Command to run this app using uvicorn:
# uvicorn app:app --reload --port 8000

```

**Comment by harupy** [comment]
@ajinkyavbhandare Thanks for the contribution!

**Comment by ajinkyavbhandare** [comment]
@harupy Thank you , I'm grateful for the opportunity to be a part of this project.

**Pull request merged**